sklearn/utils/estimator_checks.py

```diff
@@ -396,6 +396,10 @@ def set_checking_parameters(estimator):
         # which is more feature than we have in most case.
         estimator.set_params(k=1)
 
+    if name in ('HistGradientBoostingClassifier',
```

Comment:
why? speed reasons?

Reply:
I'll add a comment.

There's a test that makes sure we can correctly predict a small dataset (20 samples or so). But the default for `min_samples_leaf` is 20, so we just create one-node trees which cannot give accurate predictions.

Reply:
I wonder if this default is a problem or not.

I agree that `min_samples_leaf=20` gives better accuracy on non-toy datasets. However it can be surprising to get low predictive accuracy on toy datasets that are easily separable.

We could set a more complex default value, e.g. `min_samples_leaf=(0.1, 20)` that would be interpreted as `min(ceil(0.1 * X.shape[0]), 20)` but I am afraid of the added complexity. WDYT?

Reply:
I'd be OK to implement this as `'auto'` rather than allowing a tuple to be passed.

Let me know.

In any case I'll add a comment in the parameter docstring indicating that this value is too high for small datasets.

@Laurae2 I've been looking at your great lightGBM / xgboost parameter comparison tool (https://sites.google.com/view/lauraepp/parameters), maybe you have some suggestion?

Reply:
@NicolasHug I would recommend an `'auto'` parameter which would throw a warning if it is `'auto'` and the number of samples is too low vs the `'auto'` preset value.

Another possibility is to lower the `'auto'` value automatically when it is below the default value of `'auto'`, and also throw a warning.

If not `'auto'`, don't throw a warning (intentional from the user, you may still throw another warning if the value is higher than the number of samples).

What's your opinion on these ideas?

Reply:
I don't think auto should raise a warning since auto basically means "I trust scikitlearn to do the right thing"

Do you think @ogrisel solution of using min(ceil(0.1 * X.shape[0]), 20) is a nice default?